Webhooks/Edify.ashx

```diff
@@ -0,0 +1,455 @@
+                    .GetComponentByEntityId( communicationGuid, communicationRecipient.CommunicationId, communicationRecipient.Communication.Subject );
+
+                rockContext.SaveChanges();
+
+                var interactionService = new InteractionService( rockContext );
+                DateTime timeStamp = RockDateTime.ConvertLocalDateTimeToRockDateTime( new DateTime( 1970, 1, 1, 0, 0, 0, DateTimeKind.Utc ).AddSeconds( edifyEvent.Timestamp.Value ).ToLocalTime() );
+
+                switch ( edifyEvent.Event )
+                {
+                    case "processed":
+                        // Do nothing.
+                        break;
+                    case "dropped":
+                    case "MessageHeld":
+                    case "MessageDeliveryFailed":
+                        communicationRecipient.Status = CommunicationRecipientStatus.Failed;
+                        communicationRecipient.StatusNote = edifyEvent.Payload.Details + edifyEvent.Payload.Output;
+
+                        if ( edifyEvent.Payload.Details.Contains( "Bounced Address" ) || edifyEvent.Payload.Output.Contains( "Bounced Address" ) )
+                        {
+                            Rock.Communication.Email.ProcessBounce(
+                                edifyEvent.Payload.Message.To,
+                                Rock.Communication.BounceType.HardBounce,
+                                edifyEvent.Payload.Details,
+                                timeStamp );
+                        }
+                        else if ( edifyEvent.Payload.Status == "HardFail" )
+                        {
+                            Rock.Communication.Email.ProcessBounce(
+                                edifyEvent.Payload.Message.To,
+                                Rock.Communication.BounceType.HardBounce,
+                                edifyEvent.Payload.Output,
+                                timeStamp );
+                        }
+                        else
+                        {
+                            Rock.Communication.Email.ProcessBounce(
+                                edifyEvent.Payload.Message.To,
+                                Rock.Communication.BounceType.SoftBounce,
+                                communicationRecipient.StatusNote,
+                                RockDateTime.ConvertLocalDateTimeToRockDateTime( new DateTime( 1970, 1, 1, 0, 0, 0, DateTimeKind.Utc ).AddSeconds( edifyEvent.Timestamp.Value ).ToLocalTime() ) );
+                        }
+                        break;
+                    case "delivered":
+                    case "MessageSent":
+                        communicationRecipient.Status = CommunicationRecipientStatus.Delivered;
+                        communicationRecipient.StatusNote = string.Format( "Confirmed delivered by Edify at {0}", timeStamp.ToString() );
+                        break;
+                    case "MessageDelayed":
+                        // TODO: handle MessageDelayed.
```

Comment:
Does `CommunicationRecipientStatus.Pending` have anything else attached to it? Like actually sending a message.

Reply:
Did a quick search and I am not comfortable with setting it to Pending based on what I saw. Though I didn't confirm that is how it sends there are at least a couple methods that look for pending communications that then call a send method. So we will just leave it for now.